lib/src/values/logic_value.dart

```diff
@@ -761,6 +769,29 @@ abstract class LogicValue {
       // ignore: avoid_dynamic_calls
       _doCompare(other, (a, b) => (a >= b) as bool);
 
+  /// Power operation
+  ///
+  /// Both inputs [base] and [exponent] are either of type [int] or [BigInt].
+  /// Returns [base] raise to the power [exponent] of same input type else
+  /// it will throw an exception.
+  dynamic _powerOperation(dynamic base, dynamic exponent) {
+    if ((base is int) && (exponent is int)) {
+      return math.pow(base, exponent);
+    }
+    if ((base is BigInt) && (exponent is BigInt)) {
+      if (base == BigInt.one) {
+        return BigInt.one;
+      } else if (base == BigInt.zero && exponent > BigInt.zero) {
+        return BigInt.zero;
+      } else if (!exponent.isValidInt) {
+        throw Exception("""
```

Comment:
We are gradually trying to make ROHD throw more specific exceptions instead of `Exception`.  Can we add a new type of exception in the `exceptions/logic_value/` directory covering this flavor of exception?  It doesn't have to be super-specific to this exact function if you can come up with a more generic one (e.g. if you want to make one that a LogicValue operation couldn't be performed with a configurable message for why)